### PR TITLE
IDENTITY-6611: Add unit test cases for scim2 common listener

### PR DESCRIPTION
**Pull request by ashensw**

### CODE REVIEW

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListenerTest.java [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        scimUserOperationListener = spy(new SCIMUserOperationListener());
+        mockStatic(UserCoreUtil.class);
+        mockStatic(SCIMCommonUtils.class);
+    }
+
+    @AfterMethod
+    public void tearDown() throws Exception {
```
Comment (nilasini) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListenerTest.java]: You can remove this method

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListenerTest.java [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+    public Object[][] testGetExecutionOrderIdData() {
+
+        return new Object[][]{
+                {10, 10},
+                {IdentityCoreConstants.EVENT_LISTENER_ORDER_ID, 90}
+        };
+    }
+
+    @Test(dataProvider = "testGetExecutionOrderIdData")
+    public void testGetExecutionOrderId(int orderID, int expectedResult) throws Exception {
```
Comment (nilasini) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListenerTest.java]: add a blank line

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListenerTest.java [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+        when(userStoreManager.isSCIMEnabled()).thenReturn(isSCIMEnabled);
+        when(groupDAO.isExistingGroup(anyString(), anyInt())).thenReturn(isGroupExisting);
+        whenNew(GroupDAO.class).withNoArguments().thenReturn(groupDAO);
+        when(UserCoreUtil.getDomainName((RealmConfiguration) anyObject())).thenReturn(domainName);
+        when(UserCoreUtil.addDomainToName(anyString(), anyString())).thenReturn("testRoleNameWithDomain");
+        when(SCIMCommonUtils.getGroupNameWithDomain(anyString())).thenReturn("testRoleNameWithDomain");
+
+        assertTrue(scimUserOperationListener.doPostAddRole(roleName, userList, permissions, userStoreManager));
+    }
+
```
Comment (nilasini) [inline components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListenerTest.java]: If possible integrate this two methods(testDoPostAddRole,testDoPreDeleteRole) in to one or define a common method and use that

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListenerTest.java (hunk 2 above)
Comment (ashensw) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListenerTest.java]: As for our new convention, we have decided to remove the newline after the method signature.

File: components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListenerTest.java (hunk 3 above)
Comment (ashensw) [reply components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListenerTest.java]: Updated the PR.

**Pull request merged**